src/Aspire.Hosting.Azure.Storage/AzureStorageResource.cs

```diff
@@ -23,9 +23,9 @@ public class AzureStorageResource(string name, Action<AzureResourceInfrastructur
     private EndpointReference EmulatorQueueEndpoint => new(this, "queue");
     private EndpointReference EmulatorTableEndpoint => new(this, "table");
 
-    internal AzureBlobStorageResource? BlobStorageResource { get; set; }
-    internal AzureQueueStorageResource? QueueStorageResource { get; set; }
-    internal AzureTableStorageResource? TableStorageResource { get; set; }
+    internal IResourceBuilder<AzureBlobStorageResource>? BlobStorageBuilder { get; set; }
```

Comment:
(nit) It is kind of odd to hold onto an IResourceBuilder from a Resource. I don't see a huge deal with it, but just calling it out as different.

If we wanted to fix it, the GetBlobService() would need to create a new IResourceBuilder each time it is called, which wouldn't be terrible - they are small.